lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -55,6 +59,7 @@ def compute(self, student, course_structure, scores_client, submissions_scores):
             # self.scores may hold outdated data, force it to refresh on next access
             lazy.invalidate(self, 'scores')
```

Comment:
I'm not sure I follow - if we invalidate the lazy attribute first, then compute updated scores, it's possible for a different request to have accessed the property (and "set" it to that value) before computation is finished. We have to invalidate after the computation is finished to ensure correctness, right? That was the reasoning behind using a try-finally here.